hledger-lib/Hledger/Data/Timeclock.hs

```diff
@@ -47,47 +48,88 @@ instance Read TimeclockCode where
     readsPrec _ ('O' : xs) = [(FinalOut, xs)]
     readsPrec _ _ = []
 
+data Session = Session
+    { in' :: TimeclockEntry,
+      out :: TimeclockEntry
+    }
+
+data Sessions = Sessions
+    { completed :: [Session],
+      active :: [TimeclockEntry]
+    }
+
+-- Find the relevant clockin in the actives list that should be paired with this clockout.
+-- If there is a session that has the same account name, then use that.
+-- Otherwise, if there is an active anonymous session, use that.
+-- Otherwise, raise an error.
```

Comment:
Nice noting of the error raising, let's do that in pairClockEntries' haddock as well.

Reply:
And timeclockEntriesToTransactions.